Sources/Sugar/Validators/StrongPassword.swift

```diff
@@ -1,30 +1,47 @@
 import Vapor
 import Validation
 
-private let regex = "^.*(?=.{3,})(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[\\d\\X])(?=.*[^a-zA-Z\\d\\s:]).*$"
-
 /*The password contains characters from at least three of the following four categories:
 
  English uppercase characters (A – Z)
  English lowercase characters (a – z)
  Base 10 digits (0 – 9)
  Non-alphanumeric (For example: !, $, #, or %)
- Unicode characters
+
  */
 
 public struct StrongPassword: Validator {
+
+    static let lowerCaseRegex = "^.*[a-z].*$"
+    static let upperCaseRegex = "^.*[A-Z].*$"
+    static let digitsRegex = "^.*[0-9].*$"
+    static let specialCharRegex = "^.*[^a-zA-Z0-9].*$"
+
+    static let standardRegexes = [lowerCaseRegex, upperCaseRegex, digitsRegex, specialCharRegex]
+
     internal let minLength: Int
+    internal let minMatchingRegexes: Int
+    internal let regexes: [String]
+
+    public init(minLength: Int = 6,
+                regexes: [String] = standardRegexes,
+                minMatchingRegexes: Int = 3) {
```

Comment:
we're using the following style of splitting long function signatures:

```swift
public init(
    minLength: Int = 6,
    regexes: [String] = standardRegexes,
    minMatchingRegexes: Int = 3
) {
```

Reply:
Xcode auto-formats like this:
```swift
public init(
        minLength: Int = 6,
        regexes: [String] = standardRegexes,
        minMatchingRegexes: Int = 3
        ) {
...
}
```

Or

```swift
public init(
        minLength: Int = 6,
        regexes: [String] = standardRegexes,
        minMatchingRegexes: Int = 3)
{
...
}
```

I went with the second option, ok?

Reply:
Yeah, Xcode doesn't help much here. I looked through our API guide and noticed we don't have anything about this so this is fine. I personally stick to the style I presented, and I think @steffendsommer does too, correct?

Reply:
Yes, I think the team in general sticks to that style. I noted it down, to get something like this into our style guide.

Reply:
The IDE style doesn't really go well with our line length requirement and unfortunately you can't change the formatting in Xcode (yet).

Reply:
@cweinberger we had an example for how to format a `guard` statement, and I've now added an example for a function: https://github.com/nodes-vapor/readme/blob/master/Documentation/guide-how-to-write-swift.md#spacing